### PR TITLE
capz: add periodic and postsubmit e2e-full jobs

### PR DESCRIPTION
**Pull request by cpanato**
This PR adds:
- E2E-full periodic job
- E2E-full postsubmit job

/assign @CecileRobertMichon

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml [hunk 1]
```diff
@@ -99,3 +99,38 @@ periodics:
         privileged: true
   annotations:
     testgrid-alert-email: k8s-infra-staging-cluster-api-azure@kubernetes.io
+- name: periodic-cluster-api-provider-azure-e2e-full
+  decorate: true
+  decoration_config:
+    timeout: 4h
+  interval: 12h
```
Comment (cpanato) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml:106]: made 12 hours, but we can reduce
do we know how long take to run the full?

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml:106]: https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-azure-e2e-full looks like it takes around 1h so 2h timeout should be enough

Comment (cpanato) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml:106]: update the timeout to 2h

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**